src/app/Library/CrudPanel/Traits/Update.php

```diff
@@ -112,13 +112,13 @@ private function getModelAttributeValueFromRelationship($model, $field)
             case 'BelongsToMany':
             case 'MorphToMany':
                 $relationModels = $relatedModel->{$relationMethod};
-                $result = [];
+                $result = collect();
```

Comment:
Hey @tabacitu .

This was introduced only in 4.2 so I don't think it breaks anything the way I did it. This is the code in 4.1:

```php
    private function getModelAttributeValue($model, $field)
    {
        if (isset($field['entity'])) {
            $relational_entity = $this->parseRelationFieldNamesFromHtml([$field])[0]['name'];

            $relation_array = explode('.', $relational_entity);

            $relatedModel = $relatedModel = array_reduce(array_splice($relation_array, 0, -1), function ($obj, $method) {
                return $obj->{$method} ? $obj->{$method} : $obj;
            }, $model);

            $relationMethod = Arr::last($relation_array);

            if (method_exists($relatedModel, $relationMethod) && $relatedModel->{$relationMethod}() instanceof HasOne) {
                return $relatedModel->{$relationMethod}->{Arr::last(explode('.', $relational_entity))};
            } else {
                return $relatedModel->{$relationMethod};
            }
        }

        if (is_string($field['name'])) {
            return $model->{$field['name']};
        }

        if (is_array($field['name'])) {
            $result = [];
            foreach ($field['name'] as $key => $value) {
                $result = $model->{$value};
            }

            return $result;
        }
    }
```

So for the `Many` relations it would return an `Database\Collection` with `$relatedModel->{$relationMethod}`, I changed to send a `Support\Collection`, but accounted for that using the `is_a` php function. The reason why we can't directly return the `Database\Collection` is because we need to add the fake fields in the returned results before returning them.

Let me know if I made myself clear here. 

Pedro

Reply:
Hmm... before this commit I could open the Update page on an Invoice:
![Screenshot 2022-02-01 at 12 52 43](https://user-images.githubusercontent.com/1032474/151955913-03d9230a-7b8a-4a51-8f91-5b84c567df5e.png)

After this commit, I cannot:
![Screenshot 2022-02-01 at 12 53 35](https://user-images.githubusercontent.com/1032474/151955956-fb0d2e6d-11e9-4bde-8ab2-9bd94f5e6a0f.png)

So it's definitely breaking stuff that was previously working (in 4.2).

Reply:
Indeed @tabacitu, sorry when I said breaking previously I meant Breaking Changes. Repeatable can be part of the `Many` group of fields when used as creatable entries, forgot about that one. I will have a look, but it's just apply the same sort to the collection instead of the array. 

This is also new functionality we added in 4.2 so not breaking (BC) nothing here.

Reply:
No, please don't go modifying yet another file for this to work. Instead, please undo the changes you've made last night about this.

If you want this collection change (which will mean changing x files), please submit a different PR with it towards this branch, explaining why it's needed. We can talk about it there, think of the implications on all fields, analyze if it should be done or not.

The fact is:
- yesterday we had this branch working and 90% of files reviewed;
- today it's not working, and only 60% of the files are reviewed;

So please let's get this branch back to a working state, so we can merge it. That is our no 1 priority. If there are minor issues to be addressed, we can address them in separate PRs.